examples/tv-app/linux/include/application-launcher/ApplicationLauncherManager.cpp

```diff
@@ -59,42 +59,43 @@ ApplicationLaunchResponse ApplicationLauncherManager::proxyApplicationLaunchRequ
+                                               ZCL_LAUNCH_APP_RESPONSE_COMMAND_ID,
+                                               (chip::app::CommandPathFlags::kEndpointIdValid) };
+    chip::TLV::TLVWriter * writer          = nullptr;
+    SuccessOrExit(err = command->PrepareCommand(cmdParams));
+    VerifyOrExit((writer = command->GetCommandDataElementTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(err = writer->Put(chip::TLV::ContextTag(0), response.status));
+    SuccessOrExit(err = writer->PutString(chip::TLV::ContextTag(1), response.data.c_str()));
+    SuccessOrExit(err = command->FinishCommand());
+exit:
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(Zcl, "Failed to send %s. Error:%s", responseName, chip::ErrorStr(status));
+        ChipLogError(Zcl, "Failed to send LaunchAppResponse. Error:%s", chip::ErrorStr(err));
     }
 }
 
 EmberAfApplicationLauncherApp getApplicationFromCommand(EmberAfClusterCommand * cmd, unsigned char * reqestData,
-                                                        uint8_t * requestApplication)
+                                                        unsigned short catalogVendorId, unsigned char * applicationId)
 {
     string reqestDataString(reinterpret_cast<char *>(reqestData));
 
-    uint16_t extensionFieldSetsLen =
-        static_cast<uint16_t>(cmd->bufLen - (cmd->payloadStartIndex + emberAfStringLength((uint8_t *) reqestData) + 1));
-    uint16_t catalogVendorId = emberAfGetInt16u(requestApplication, 0, extensionFieldSetsLen);
-    uint8_t * applicationId  = emberAfGetString(requestApplication, 0, extensionFieldSetsLen);
-
     EmberAfApplicationLauncherApp application = {};
     application.applicationId                 = applicationId;
     application.catalogVendorId               = catalogVendorId;
     return application;
 }
-
-bool emberAfApplicationLauncherClusterLaunchAppCallback(unsigned char * reqestData, uint8_t * requestApplication)
+bool emberAfApplicationLauncherClusterLaunchAppCallback(chip::app::Command * command, unsigned char * requestData,
+                                                        unsigned short requestApplicationCatalogVendorId,
+                                                        unsigned char * requestApplicationId)
 {
-    EmberAfApplicationLauncherApp application = getApplicationFromCommand(emberAfCurrentCommand(), reqestData, requestApplication);
-    string reqestDataString(reinterpret_cast<char *>(reqestData));
-
+    EmberAfApplicationLauncherApp application =
+        getApplicationFromCommand(emberAfCurrentCommand(), requestData, requestApplicationCatalogVendorId, requestApplicationId);
+    string reqestDataString(reinterpret_cast<char *>(requestData));
     ApplicationLaunchResponse response = ApplicationLauncherManager().proxyApplicationLaunchRequest(application, reqestDataString);
-
-    sendResponse("LaunchAppResponse", response);
+    sendResponse(command, response);
```

Comment:
So what I would expect is that the shared cluster logic in `src/app/clusters` would do things like "call an app-provided method to get the ApplicationLaunchResponse" and "send the response", and then the app would do the rest as needed....

Reply:
Updated. Now shared cluster logic has these "proxyMethods" to example apps. 
I've used naming convention of `clusterNameClusterCommandName()` for methods.

Reply:
Thank you!